### PR TITLE
LIMS-3501

### PR DESCRIPTION
**Pull request by SalmaaRagab**

### CODE REVIEW

**Comment by SalmaaRagab** [comment]
**Pass**

File: ui_testing/pages/testplan_page.py [hunk 1]
```diff
@@ -143,4 +148,12 @@ def save_and_confirm_popup(self):
...
\ No newline at end of file
+        self.sleep_small()
+
+    def delete_all_testunits(self):
+        testunits_still_available = 1
+        while testunits_still_available:
+            try:
+                self.base_selenium.click(element='test_plan:remove_testunit')
+            except:
+                testunits_still_available = 0
```
Comment (0xIslamTaha) [inline ui_testing/pages/testplan_page.py]: not the best solution. Maybe u can use delete_first_test_unit method here

File: ui_testing/pages/testplan_page.py [hunk 2]
```diff
@@ -46,7 +46,12 @@ def set_test_plan(self, name='', random=False):
 
     def set_test_unit(self, test_unit='', **kwargs):
         self.base_selenium.click('test_plan:next')
-        self.base_selenium.click('test_plan:add_test_units')
+        try:
+            self.base_selenium.click(element='test_plan:add_test_units_first_column')
+        except NoSuchElementException:
+            print('here')
+            self.base_selenium.click(element='test_plan:add_test_units_second_column')
```
Comment (0xIslamTaha) [inline ui_testing/pages/testplan_page.py]: doesn't make sense for me

File: ui_testing/pages/testplans_page.py [hunk 3]
```diff
@@ -36,3 +36,9 @@ def get_testunits_in_testplans(self, test_plan_name=''):
     
         return testplan_testunits
 
+    def get_testplan_version_and_status(self, search_text=''):
```
Comment (0xIslamTaha) [inline ui_testing/pages/testplans_page.py]:
```suggestion
    def get_testplan_version_and_status(self, search_text):
```

File: ui_testing/testcases/basic_tests/test05_testplans.py [hunk 4]
```diff
@@ -54,5 +54,35 @@ def test001_test_plan_delete_testunit(self):
         deleted_test_unit_found = self.test_plan.check_if_deleted_testunit_is_available(all_testunits=all_testunits, deleted_test_unit=deleted_test_unit)
         
         self.assertFalse(deleted_test_unit_found)
+        
+    def test004_test_plan_completed_to_completed(self):
+        '''
+        LIMS-3501
+        When the testplan status is converted from completed to completed a new version is created
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test05_testplans.py]: from completed to completed?

File: ui_testing/testcases/basic_tests/test05_testplans.py [hunk 5]
```diff
@@ -54,5 +54,35 @@ def test001_test_plan_delete_testunit(self):
...
+            self.test_plan.save_and_confirm_popup()
+
+            # go back to the active table
+            self.test_plan.get_test_plans_page()
 
+            # get the testplan to check its version
+            self.base_selenium.LOGGER.info('Getting the currently changed testplan to check its status and version')
+            inprogress_testplan_version, testplan_row_data_status = self.test_plan.get_testplan_version_and_status(search_text=old_completed_testplan_name)
 
+            self.assertEqual(old_completed_testplan_version + 1, int(inprogress_testplan_version))
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test05_testplans.py]:
```suggestion
            self.assertGreater(int(inprogress_testplan_version), old_completed_testplan_version)
```

**Pull request closed without merging**